src/Common/src/TypeSystem/IL/Stubs/ArrayMethodILEmitter.cs

```diff
@@ -147,6 +147,17 @@ private void EmitILForAccessor()
 
             MethodDesc throwHelper = _method.Context.GetHelperEntryPoint("ArrayMethodILHelpers", "ThrowIndexOutOfRangeException");
             codeStream.Emit(ILOpcode.call, _emitter.NewToken(throwHelper));
+
+            if (_method.Kind == ArrayMethodKind.Address || _method.Kind == ArrayMethodKind.Get)
+            {
+                // Throw helper won't return, but we still need to push something on the stack
+                // for method kinds that have a return value.
+                var returnValue = _emitter.NewLocal(_elementType);
```

Comment:
I wasn't sure either, but RyuJIT didn't complain.
But I'll make it an infinite loop. Sounds like a better plan.